packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,219 @@
+    this.decoration =
+        const BoxDecoration(color: CupertinoColors.secondarySystemBackground),
+  })  : _type = _CupertinoFormSectionType.centered,
+        super();
+
+  final _CupertinoFormSectionType _type;
+
+  /// Sets the form section title.
+  final String? title;
+
+  /// Sets padding for the section title.
+  ///
+  /// Defaults to the standard iOS padding, which is determined
+  /// using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry titlePadding;
+
+  /// Padding for the the [Column] encapsulating the [children].
+  ///
+  /// Defaults to zero padding if constructed with standard
+  /// [CupertinoFormSection] constructor. Defaults to the standard notched-style
+  /// iOS padding when constructing with [CupertinoFormSection.centered].
+  /// Determined using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry rowsPadding;
+
+  /// The list of rows in the section.
+  final List<Widget> children;
+
+  /// Sets the decoration for the section.
+  ///
+  /// Defaults to a [CupertinoColors.secondarySystemBackground] background
+  /// color.
+  final BoxDecoration decoration;
+
+  /// Sets the circular border radius for the [Column] encapsulating [children]
+  /// rows.
+  ///
+  /// Defaults to 0.0 when constructed with standard [CupertinoFormSection]
+  /// constructor. Defaults to 10.0 when constructing with
+  /// [CupertinoFormSection.centered].
+  final double borderRadius;
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoFormSectionState();
+}
+
+class _CupertinoFormSectionState extends State<CupertinoFormSection> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const Divider longDivider = Divider(
+      endIndent: 0,
+      indent: 0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    const Divider shortDivider = Divider(
+      endIndent: 0,
+      indent: 15.0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    final List<Widget> children =
+        (widget._type == _CupertinoFormSectionType.base)
+            ? <Widget>[longDivider]
+            : <Widget>[];
+
+    widget.children
+        .sublist(0, widget.children.length - 1)
+        .forEach((Widget element) {
+      children.add(element);
+      children.add(shortDivider);
+    });
+
+    children.add(widget.children.last);
+    if (widget._type == _CupertinoFormSectionType.base) {
+      children.add(longDivider);
+    }
+
+    return DecoratedBox(
+      decoration: const BoxDecoration(
+        color: CupertinoColors.secondarySystemBackground,
+      ),
+      child: Column(
+        children: <Widget>[
+          Padding(
+            padding: widget.titlePadding,
+            child: Row(
+              children: <Widget>[
+                Text(
+                  widget.title!.toUpperCase(),
```

Comment:
We don't have to force users to upper case their text. We can just recommend it in the doc. Also if title/header was a widget, you can then just insert it here. 

You can also use multiple layers of paddings. You can just have a general 16px padding that's internal when you use the padded version of the constructor. Then you might not need a title padding. If the user wants some sort of padding, they can just compose it themselves in that widget. You can then just add the default 10px vertical padding internally in the column too. This way, you don't need a titlePadding field.